documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/server/http/HTTPServerDocs.java

```diff
@@ -876,7 +878,7 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         shopContext.setHandler(new ShopHandler());
 
         // You want to gzip the shop web application only.
-        GzipHandler shopGzipHandler = new GzipHandler();
+        DynamicCompressionHandler shopGzipHandler = ServiceLoader.load(DynamicCompressionHandler.class).iterator().next();
```

Comment:
I can't see the benefit of using the `ServiceLoader` this way. A user can just extend `GzipHandler` and then call `ServletContextHandler.setGzipHandler`.

Can you explain the problem you're trying to solve a bit more?

Reply:
specific approach inspired by original issue https://github.com/eclipse/jetty.project/issues/8769 but personally i think it's okay to have it either way as long as interface is consistent